add UI for certifying organisation create and details page

ann26 opened the pull request:
fix #398 

*certification menu*
![screenshot from 2017-04-17 20-18-58](https://cloud.githubusercontent.com/assets/26101337/25117331/d5434c52-243b-11e7-9f58-75341d1bfe68.png)

*certifying organisation create* **update**
![screenshot from 2017-05-05 17-59-13](https://cloud.githubusercontent.com/assets/26101337/25743179/4627fb64-31bd-11e7-8a45-e3172073e1f5.png)
![screenshot from 2017-05-05 18-13-45](https://cloud.githubusercontent.com/assets/26101337/25743476/ce4b23ee-31be-11e7-8564-7860daeab027.png)


*no certifying organisation are defined*
![screenshot from 2017-04-18 13-38-25](https://cloud.githubusercontent.com/assets/26101337/25117443/63f1b0b0-243c-11e7-8093-643f4dafb0db.png)

*certifying organisation lists - this page show up is when a user hit organisation button from certification menu*
*view for organisation owner and staff user*
![screenshot from 2017-04-19 13-29-11](https://cloud.githubusercontent.com/assets/26101337/25166163/3500cd18-2504-11e7-8292-79e8139277f3.png)
*view for other user*
![screenshot from 2017-04-20 15-58-58](https://cloud.githubusercontent.com/assets/26101337/25222311/54753c04-25e2-11e7-959e-d54fc5e9f2d2.png)

*certifying organisation details* 
*view for organisation owners and staff users*
![screenshot from 2017-04-20 16-02-08](https://cloud.githubusercontent.com/assets/26101337/25222521/e380b018-25e2-11e7-92ef-72a293abad91.png)
![screenshot from 2017-04-20 16-02-20](https://cloud.githubusercontent.com/assets/26101337/25222522/e49f98d8-25e2-11e7-9987-b8bcba1c1b0f.png)
![screenshot from 2017-04-20 16-02-51](https://cloud.githubusercontent.com/assets/26101337/25222529/e6fb7430-25e2-11e7-92f8-f8584bb81410.png)
![screenshot from 2017-04-20 16-03-02](https://cloud.githubusercontent.com/assets/26101337/25222538/e89986ec-25e2-11e7-84fa-42410a12241a.png)

*view for user who is a course convener - they can only edit course*
![screenshot from 2017-04-21 14-17-12](https://cloud.githubusercontent.com/assets/26101337/25266956/3f829a1a-269d-11e7-8f38-0630465767ce.png)
![screenshot from 2017-04-21 14-11-26](https://cloud.githubusercontent.com/assets/26101337/25266836/de5b507e-269c-11e7-8275-0c192bf44b93.png)
![screenshot from 2017-04-21 14-16-02](https://cloud.githubusercontent.com/assets/26101337/25266911/1df692ca-269d-11e7-8743-72fd73cc6530.png)
![screenshot from 2017-04-21 14-16-14](https://cloud.githubusercontent.com/assets/26101337/25266912/1f4738a0-269d-11e7-8181-167774aca644.png)

*view for other users*
![screenshot from 2017-04-20 15-59-51](https://cloud.githubusercontent.com/assets/26101337/25222388/8b5398ce-25e2-11e7-81ff-68ec368a8347.png)
![screenshot from 2017-04-20 16-00-05](https://cloud.githubusercontent.com/assets/26101337/25222392/8e131d00-25e2-11e7-9195-4b50f8010273.png)
![screenshot from 2017-04-20 16-00-18](https://cloud.githubusercontent.com/assets/26101337/25222400/8faa2c26-25e2-11e7-83e3-286832704ecd.png)
![screenshot from 2017-04-20 16-00-29](https://cloud.githubusercontent.com/assets/26101337/25222403/91362176-25e2-11e7-8dda-72eb82d3b852.png)


*when no training center, course types, course, course convener are defined*
![screenshot from 2017-04-21 14-44-51](https://cloud.githubusercontent.com/assets/26101337/25268008/49f303b4-26a1-11e7-9c45-3e7bdee57d5f.png)
![screenshot from 2017-04-21 14-45-12](https://cloud.githubusercontent.com/assets/26101337/25268009/4ab3e1ba-26a1-11e7-96e7-164d418b23f4.png)
![screenshot from 2017-04-21 14-45-32](https://cloud.githubusercontent.com/assets/26101337/25268012/4c044b4a-26a1-11e7-8d13-02595f505d64.png)

*delete view*
![screenshot from 2017-04-19 16-43-05](https://cloud.githubusercontent.com/assets/26101337/25173713/5a4942b0-251f-11e7-9604-0c3f4f0317b2.png)

timlinux commented:
@geomenke FYI

timlinux commented:
Looking good @ann26 here are some notes:

![screen shot 2017-04-21 at 1 25 19 pm](https://cloud.githubusercontent.com/assets/178003/25275527/0a9c405a-2696-11e7-82dd-973d6b0444d8.png)

Can you please use a better widget for selecting owners so that you can type and the name list gets filtered to show matches. Also they should be listed alphabetically.

timlinux commented:
![screen shot 2017-04-21 at 1 33 05 pm](https://cloud.githubusercontent.com/assets/178003/25275794/5e06f96e-2697-11e7-9152-c21d5560e505.png)

Clicking on the buttons here does not do anything for me.

ann26 commented:
@timlinux thanks, I'll look into it.
for those buttons, they are not working here, they will be made in separate pull requests

meomancer commented:
Note :
this PR scope just

- [ ] Create certifying organization
- [ ] Accept/reject pending certifying organization
- [ ] Display detail

My note maybe:
1. Make validation for email, now we could input non-email format
2. Found bug on organisation (i'm not sure with others model)
So basically, we can create same name of organization. It will gives same slug both of them, and when we open the detail of organization or maybe delete one of organization, it will error, returned 2. 
how to produce:
- create 2 cert organization with same name
- go to one of organization

Solution maybe :
- Make name of organization uniquer
or
- Update save method, create method to generate slug, for example make slug 'organization-1', 'organization-2' if there are cert org that already named 'organization'

3. Not sure what map tells about in detail. It is location of organization? 
- Maybe we need to call google map api to get location based on address and country, and we safe the point that returned from google.
or
-Maybe no need map location of organization?

what do you think @timlinux ? for number 2 & 3

ann26 commented:
thanks @meomancer 
I updated the code and added the email validators
![screenshot from 2017-05-02 13-37-15](https://cloud.githubusercontent.com/assets/26101337/25606617/92651a3a-2f3c-11e7-8762-ef556a5c256f.png)

meomancer commented on django_project/certification/models/certifying_organisation.py:
```diff
@@ -59,6 +62,16 @@ def get_queryset(self):
+        validate_email(value)
+        return True
+    except ValidationError(
+            _('%(value)s is not proper gitter room name'),
```
could you change the value error @ann26 ?
is not a valid email?

ann26 replied:
Ah yes sure

timlinux commented:
@meomancer 

For 2: I like the scheme of organisation-1, organisation-2 for the slug but maybe just validating as unique is enough because it is extremely unlikely that we have two companies with the same name.

For 3: Location shows where the training centers are - so eventually we can make a map of all training centers in the world. I think we don't need to map the organization itself other than to the country level. I'm ok with just letting the user click the position on the map for now in the training center form - its a very low frequency activity. If you have the geocoding stuff from feet easily able to be cut & pasted into the project codebase we can use that but otherwise lets just let the user do it manually by clicking the place on the map.

ann26 commented:
Hi @timlinux , so travis has made updates after may 4th (https://blog.travis-ci.com/2017-05-04-precise-image-updates) , 
which makes my recent commits failed in travis check. I did some experiment with the travis yml file in #411 and it turns out I had to add `sudo: false` in the file. However, I am not sure if this change is good. What do you think?

lucernae commented:
I don't know if this will help:
https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments

It might be because previously, the default for this repository (when sudo is not explicitly stated in .travis.yml) is sudo: false. But since 4th May star wars day, they default to sudo: required? then causing it to use the latest edge precise build.

timlinux commented:
@ann26 fine to change sudo if needed as long as the tests still function in the same way....

ann26 commented:
@timlinux ok

meomancer commented on django_project/certification/migrations/0004_auto_20170505_1230.py:
```diff
@@ -0,0 +1,24 @@
+        migrations.AlterField(
+            model_name='certifyingorganisation',
+            name='address',
+            field=models.TextField(help_text=b'Address of Organisation or Institution.', max_length=1000),
```
Use ugettext_lazy for translation, but i think we could make another tickets.

meomancer commented:
LGTM, for me
